gpkit/nomials.py

```diff
@@ -591,97 +639,174 @@ def __init__(self, left, right, oper_ge=True):
+        self.p_lt, self.m_gt = p_lt, m_gt
+        self.posylt1_rep = p
+        self.substitutions = p.values
+
+    def as_posyslt1(self):
+        posys = listify(self.posylt1_rep)
+        if not self.substitutions:
+            # just return the pre-generated posynomial representation
+            return posys
+
+        out = []
+        for posy in posys:
+            if hasattr(self, "m_gt"):
+                m_gt = self.m_gt.sub(self.substitutions,
+                                     require_positive=False)
+                if m_gt.c == 0:
+                    return []
+
+            _, exps, cs, _ = substitution(posy, self.substitutions)
+            # remove any cs that are just nans and/or 0s
+            nans = np.isnan(cs)
+            if np.all(nans) or np.all(cs[~nans] == 0):
+                return []  # skip nan'd or 0'd constraint
+
+            exps, cs, pmap = simplify_exps_and_cs(exps, cs, return_map=True)
+
+            #  The monomial sensitivities from the GP/SP are in terms of this
+            #  smaller post-substitution list of monomials, so we need to map
+            #  back to the pre-substitution list.
+            #
+            #  A "pmap" is a list of HashVectors (mmaps), whose keys are
+            #  monomial indexes pre-substitution, and whose values are the
+            #  percentage of the simplified  monomial's coefficient that came
+            #  from that particular parent.
+
+            self.pmap = pmap
+            p = Posynomial(exps, cs, simplify=False)
+            if p.any_nonpositive_cs:
+                raise RuntimeWarning("PosynomialConstraint %s became Signomial"
+                                     " after substitution" % self)
+            out.append(p)
+        return out
+
+    def sens_from_dual(self, p_senss, m_sensss):
+        if not p_senss or not m_sensss:
+            # as_posyslt1 created no inequalities
+            return {}, {}
+        p_sens, = p_senss
+        m_senss, = m_sensss
+        presub = self.posylt1_rep
+        constr_sens = {"overall": p_sens}
+        if hasattr(self, "pmap"):
+            m_senss_ = np.zeros(len(presub.cs))
+            counter = 0
+            for i, mmap in enumerate(self.pmap):
+                for idx, percentage in mmap.items():
+                    m_senss_[idx] += percentage*m_senss[i]
+            m_senss = m_senss_
+        # Monomial sensitivities
+        constr_sens[str(self.m_gt)] = p_sens
+        for i, mono_sens in enumerate(m_senss):
+            mono = Monomial(self.p_lt.exps[i], self.p_lt.cs[i])
+            constr_sens[str(mono)] = mono_sens
+        # Constant sensitivities
+        var_senss = {var: sum([presub.exps[i][var]*m_senss[i] for i in locs])
+                     for (var, locs) in presub.varlocs.items()
+                     if var in self.substitutions}
+        return constr_sens, var_senss
+
+
+class MonoEQConstraint(PosynomialConstraint):
```

Comment:
This might be carry-over from before, but I don't think `MonoEQConstraint` should inherit from `PosynomialConstraint`. According to the docstring, a `PosynomialConstraint` is "A constraint of the general form monomial >= posynomial". A `MonoEQConstraint` certainly does not have this form.